apps/angular/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,60 @@
 import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
 import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { TODO } from './app.service';
+import {
+  deleteTodoAction,
+  getAllTodosAction,
+  updateTodoAction,
+} from './store/app.actions';
+import { isError, isLoading, selectAllTodos } from './store/app.selectors';
+
+// import { GetAllTodoActions } from './store/app.actions';
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
+    @if (isError$ | async) {
+      <p class="error">{{ isError$ | async }}</p>
+    }
+    @if (isLoading$ | async) {
+      <mat-spinner></mat-spinner>
+    }
+    <div *ngFor="let todo of todos$ | async">
       {{ todo.title }}
       <button (click)="update(todo)">Update</button>
+      <button (click)="delete(todo.id)">Delete</button>
     </div>
   `,
   styles: [],
 })
 export class AppComponent implements OnInit {
-  todos!: any[];
+  todos$!: Observable<TODO[]>;
+  isError$!: Observable<string>;
+  isLoading$!: Observable<boolean>;
 
-  constructor(private http: HttpClient) {}
+  constructor(private store: Store<{ todos: TODO[] }>) {}
 
   ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        this.todos = todos;
-      });
+    this.todos$ = this.store.select(selectAllTodos);
+    this.isLoading$ = this.store.select(isLoading);
+    this.isError$ = this.store.select(isError);
```

Comment:
you can instanciate it directly inside the field initializer

And you can use selectSignal to have a signal